frontend/src/routes/login.lazy.tsx

```diff
@@ -65,18 +66,94 @@ function Login() {
             Log in
           </h1>
           <div className="card flex flex-col gap-3">
-            {data?.errors?.root && (
-              <div style={{ color: "red" }}>{data.errors.root}</div>
-            )}
-            <div>
+            <Form.Root>
+              {data?.errors && (
+                <Form.Message
+                  className="text-red-500 text-sm"
+                  match={"valueMissing"}
+                >
+                  {data.errors.root}
+                </Form.Message>
+              )}
+
+              <Form.Field className="my-5" name="username">
```

Comment:
```suggestion
              <Form.Field className="my-2" name="username">
```

Reduce the spacing between fields